.circleci/config.yml

```diff
@@ -1,832 +1,40 @@
-          context: data-eng-bigquery-etl-dockerhub
-          requires:
-            - generate-sql
-            # Public image must be pushed after the private one because of
-            # webhooks used in Ops logic. For details, see:
-            # https://bugzilla.mozilla.org/show_bug.cgi?id=1715628#c0
-            - deploy-to-private-gcr
-          filters:
-            branches:
-              only:
-                - main
-      # The following "private" jobs are basically clones of the public jobs
-      # for generate-sql, deploy, and push-generated-sql, except that they pull
-      # in some additional content from an internal Mozilla repository for
-      # cases where ETL code cannot be public. Although the CI logic is
-      # consolidated in this public repository, note that we are both pulling
-      # from the internal repository and pushing generated results back to
-      # a branch on that internal repository, which may be initially
-      # surprising.
-      - private-generate-sql:
-          filters:
-            branches:
-              only:
-                - main
-      - push-private-generated-sql:
-          requires:
-            - private-generate-sql
-          filters:
-            branches:
-              only:
-                - main
-      - sync-dags-repo:
-          name: 🔃 Synchronize private-bigquery-etl submodule
-          repo-to-sync: private-bigquery-etl
-          target-branch: private-generated-sql
-          requires:
-            - push-private-generated-sql
-          filters:
-            branches:
-              only:
-                - main
-      - deploy-to-private-gcr:
-          context: data-eng-airflow-gcr
-          requires:
-            - private-generate-sql
-            # can't run in parallel because CIRCLE_BUILD_NUM is same
-            - build
-            - generate-sql
-          filters:
-            branches:
-              only:
-                - main
-  nightly:
-    # Run after schema-generator to ensure we are up-to-date
-    triggers:
-      - schedule:
-          cron: "0 5 * * *"
-          filters:
-            branches:
-              only:
-                - main
-    jobs: *build_jobs
+  filter-paths-main:
+    when:
+      and:
+        - equal: [main, << pipeline.git.branch >>]
+    jobs:
+      - path-filtering/filter:
+          base-revision: << pipeline.git.base_revision >>
+          config-path: .circleci/ci-config.yml
+          mapping: |
+            bigquery_etl/.* validate-bqetl true
+            tests/.* validate-bqetl true
+            .* validate-sql true
+            .* validate-routines true
+  filter-paths-pr:
+    when:
+      and:
+        - not:
+            equal: [main, << pipeline.git.branch >>]
+    jobs:
+      - path-filtering/filter:
+          base-revision: main
+          config-path: .circleci/ci-config.yml
+          mapping: |
+            bigquery_etl/.* validate-bqetl true
+            tests/.* validate-bqetl true
```

Comment:
I suspect there might be some changes here that it would be good to run through the sql jobs for - changes to how we handle schemas, metadata, views, etc.